### PR TITLE
Motions and Disputes Finalize Motion Widget

### PR DESCRIPTION
**Pull request by rdig**
## Description

This PR adds in the finalize motion widget, alongside the voting results

This is used both for voting and staking, in the latter case it won't show the vote results, just the finalize button.

**Changes** 

- [x] Added `VoteResults` component
- [x] Added `FinalizeMotionWidget` component
- [x] Fixed `getMotionState` logic -- _we'll need to fix it further, but we're waiting on Product for info_
- [x] Cherry picked `MotionVote` types from here: c1ac6e635b20be784a2a3b4ce0f1281ae63ebf38
- [x] Added `motionVoteResults` resolver
- [x] Added motion domain to return values of `colonyActions` resolver

**Demo**

User side won:
demo-md-finalize-widget-1

User side lost:
demo-md-finalize-widget-2

Difference between a finalize STAKE and a finalized VOTE:
demo-md-finalize-widget-3

Resolves DEV-291

### CODE REVIEW

**Review by alicjakujawa — CHANGES_REQUESTED** [review]
Noticed small issue, if i didint vote, but i have reputation in colony, i see:
Screenshot 2021-04-21 at 14 35 25

Another one - finalize button should be enabled for every user
Screenshot 2021-04-21 at 14 40 28

**Comment by rdig** [comment]
@alicjakujawa 

> Noticed small issue, if i didn't vote, but i have reputation in colony, i see:

Fixed

> Another one - finalize button should be enabled for every user

This is actually correctly implemented, since you can only finalize if you've registered your account _(and your user in the test hadn't registered)_

File: src/data/graphql/typeDefs.ts [hunk 1]
```diff
@@ -209,6 +209,14 @@ export default gql`
     vote: Int!
   }
 
+  type MotionVoteResults {
+    currentUserVoteSide: Int!
+    yayVotes: String!
+    yayVoters: [String!]!
+    nayVotes: String!
+    nayVoters: [String]!
```
Comment (ArmandoGraterol) [inline src/data/graphql/typeDefs.ts:217]: Why is nayVoters `[String]!` but yayVoters `[String!]!`?

**Review by chinins — APPROVED** [review]
Good to go!

File: src/modules/dashboard/components/ActionsPage/FinalizeMotionWidget/VoteResults/VoteResults.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  voteYAY: {
+    id: 'dashboard.ActionsPage.FinalizeVoteWidget.VoteResults.voteYAY',
+    defaultMessage: `Yes`,
+  },
+  voteNAY: {
+    id: 'dashboard.ActionsPage.FinalizeVoteWidget.VoteResults.voteNAY',
+    defaultMessage: `No!`,
+  },
+  loadingText: {
+    id: 'dashboard.ActionsPage.FinalizeVoteWidget.VoteResults.loadingText',
```
Comment (ArmandoGraterol) [inline src/modules/dashboard/components/ActionsPage/FinalizeMotionWidget/VoteResults/VoteResults.tsx:32]: The ids of those MSGs should be `dashboard.ActionsPage.FinalizeMotionWidget.VoteResults ...`, right?

File: src/modules/dashboard/components/ActionsPage/FinalizeMotionWidget/VoteResults/VoteResults.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+  const yayVotesValue = bigNumberify(yayVotes)
+    .div(bigNumberify(10).pow(18))
+    .toNumber();
+  const nayVotesValue = bigNumberify(nayVotes)
+    .div(bigNumberify(10).pow(18))
+    .toNumber();
+  /*
+   * That moment when you have to do percentage math in plain JS because
+   * BigNumber is afraid of floating point
```
Comment (ArmandoGraterol) [inline src/modules/dashboard/components/ActionsPage/FinalizeMotionWidget/VoteResults/VoteResults.tsx:73]: Bruh

File: src/modules/dashboard/components/ActionsPage/FinalizeMotionWidget/VoteResults/VoteResultsItem.tsx [hunk 4]
```diff
@@ -20,14 +20,14 @@ interface Props {
...
   maxValue: number;
   maxPercentage?: number;
-  title: MessageDescriptor;
+  title: string | MessageDescriptor;
   voters?: Address[];
   maxAvatars?: number;
 }
 
-const displayName = 'VoteResultsWidget';
+const displayName = `dashboard.ActionPage.FinalizeVoteWidget.VoteResults.VoteResultsItem`;
```
Comment (ArmandoGraterol) [inline src/modules/dashboard/components/ActionsPage/FinalizeMotionWidget/VoteResults/VoteResultsItem.tsx:28]: Same here, should be `FinalizeMotionWidget`

**Review by ArmandoGraterol — APPROVED** [review]
It seems good to me! I just have some questions, also noticed a few tiny weeny things

File: src/modules/dashboard/components/ActionsPage/FinalizeMotionWidget/VoteResults/VoteResultsItem.tsx (hunk 4 above)
Comment (rdig) [reply src/modules/dashboard/components/ActionsPage/FinalizeMotionWidget/VoteResults/VoteResultsItem.tsx:28]: Good catch!

File: src/modules/dashboard/components/ActionsPage/FinalizeMotionWidget/VoteResults/VoteResults.tsx (hunk 2 above)
Comment (rdig) [reply src/modules/dashboard/components/ActionsPage/FinalizeMotionWidget/VoteResults/VoteResults.tsx:32]: Good catch!

File: src/data/graphql/typeDefs.ts (hunk 1 above)
Comment (rdig) [reply src/data/graphql/typeDefs.ts:217]: Good catch!

**Pull request merged**